From d683878dbeab3cd5cd073e40b41f24a4c7f62415 Mon Sep 17 00:00:00 2001
From: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Date: Tue, 15 Aug 2017 08:53:18 -0700
Subject: [PATCH] ARM: OMAP3+: PRM: fix of_irq_get() result check
Git-commit: d683878dbeab3cd5cd073e40b41f24a4c7f62415
Patch-mainline: v4.14-rc1
References: bsc#1051510

of_irq_get() may return 0 as well as a nagative error number on failure
(and never on success), however omap3xxx_prm_late_init() regards 0 as a
valid IRQ -- fix this.

Fixes: 1e037794f7f ("ARM: OMAP3+: PRM: register interrupt information from DT")
Signed-off-by: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Signed-off-by: Tony Lindgren <tony@atomide.com>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 arch/arm/mach-omap2/prm3xxx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/arm/mach-omap2/prm3xxx.c
+++ b/arch/arm/mach-omap2/prm3xxx.c
@@ -709,7 +709,7 @@ static int omap3xxx_prm_late_init(void)
 		np = of_find_matching_node(NULL, omap3_prm_dt_match_table);
 		if (np) {
 			irq_num = of_irq_get(np, 0);
-			if (irq_num >= 0)
+			if (irq_num > 0)
 				omap3_prcm_irq_setup.irq = irq_num;
 		}
 	}
